viewer-prototype/src/node/trace-server-service.ts

```diff
@@ -0,0 +1,14 @@
+import { spawn, exec } from 'child_process';
+import { injectable } from 'inversify';
+import { TraceServerConfigService } from '../common/trace-server-config';
+@injectable()
+export class TraceServerServiceImpl implements TraceServerConfigService {
+
+    async startTraceServer(path: string | undefined, port: number | undefined): Promise<void> {
+        spawn(`..${path}`, ['-vmargs', `-Dtraceserver.port=${port}`]);
+    }
+
+    async stopTraceServer(port: number | undefined): Promise<void> {
+        exec(`kill -9 $(lsof -t -i:${port})`);  // FIXME: Better approach to kill the server at a given port
```

Comment:
I guess you'll have to find a better approach now... ;-)
```
$ lsof -t -i:8080
32190
32627
$ ps aux | grep 32190
gbastien 32190 26.1  3.0 9837920 483804 tty2   Sl+  14:13   1:13 /usr/lib/firefox/firefox -new-window
$ ps aux | grep 32627
<The real trace server process>
```

Reply:
Interesting it didn't kill the browser for me (I was using chrome) and I was able to start/stop multiple times on the same tab. However, I think what you suggested (-sTCP: LISTEN) makes more sense and I have updated it.